app/src/main/java/com/github/epfl/meili/PhotoDemoActivity.kt

```diff
@@ -0,0 +1,76 @@
+package com.github.epfl.meili
+
+import android.graphics.ImageDecoder.createSource
+import android.graphics.ImageDecoder.decodeBitmap
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.MediaStore.Images.Media.getBitmap
+import android.view.View
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import com.github.epfl.meili.R
+import com.github.epfl.meili.storage.FirebaseStorageService
+import com.github.epfl.meili.storage.StorageService
+import com.squareup.picasso.Picasso
+
+class PhotoDemoActivity : AppCompatActivity() {
+
+    companion object {
+        var storageService: () -> StorageService = { FirebaseStorageService }
+    }
+
+    private lateinit var choose: Button
+    private lateinit var upload: Button
+    private lateinit var show: Button
+    private lateinit var imageView: ImageView
+    private var filePath: Uri? = null
+
+    private val getContent: ActivityResultLauncher<String> =
+        registerForActivityResult(ActivityResultContracts.GetContent()) { filePath ->
+            val bitmap = if (Build.VERSION.SDK_INT < 28) {
+                getBitmap(contentResolver, filePath)
+            } else {
+                decodeBitmap(createSource(contentResolver, filePath))
+            }
```

Comment:
Can you please explain this part? I don't understand what's the difference between build versions earlier than 28 and the ones after 28.

Reply:
Yes! In fact, the `getBitmap` function has been deprecated and replaced by `decodeBitmap` in Android. However, SDK versions < 28 don't support `decodeBitmap`. Since our app's min SDK version is 23, we need to support both old and new versions.